docs/resources/db_target.md

```diff
@@ -0,0 +1,263 @@
+[`database_authentication_config.authentication_type`](#authentication_type)
+after initial resource creation may require that the Db target is reconfigured
+for the new authentication type. For example, if you change the attribute to
+`ServiceAccountInjection` and
+[`database_authentication_config.cloud_service_provider`](#cloud_service_provider)
+to `GCP`, then you must also update the target's [`remote_host`](#remote_host)
+to include a `gcp://` protocol prefix.
+
+~> **Warning** _Modifying_ a Db target's
+[`database_authentication_config.database`](#database) after initial resource
+creation may require you to reconfigure the target's allowed target users as
+governed by BastionZero policy.
+
+### Remote host
+
+There are some combination of values in the
+[`database_authentication_config`](#database_authentication_config) attribute
+that restrict the [`remote_host`](#remote_host) attribute; namely, there are
+instances where an expected protocol prefix string must be included.
+
+- If
+[`database_authentication_config.cloud_service_provider`](#cloud_service_provider)
+is equal to `GCP`, then [`remote_host`](#remote_host) must include a `gcp://`
+protocol prefix. 
+
+- If
+[`database_authentication_config.cloud_service_provider`](#cloud_service_provider)
+is equal to `AWS` and [`database_authentication_config.database`](#database) is
+equal to `MySQL`, then [`remote_host`](#remote_host) must include an
+`rdsmysql://` protocol prefix.
+
+- If
+[`database_authentication_config.cloud_service_provider`](#cloud_service_provider)
+is equal to `AWS` and [`database_authentication_config.database`](#database) is
+equal to `Postgres`, then [`remote_host`](#remote_host) must include an `rds://`
+protocol prefix.
+
+## Example Usage
+
+### Db target via proxy target
+
+Create a Db target with the default authentication configuration
+(non-passwordless), and use a Bzero target to proxy the connection to the
+configured database.
+
+```terraform
+data "bastionzero_environments" "example" {}
+data "bastionzero_bzero_targets" "example" {}
+
+locals {
+  # Find environment with name "example-env". `env` is null if not found
+  env = one([
+    for each in data.bastionzero_environments.example.environments
+    : each if each.name == "example-env"
+  ])
+  # Find Bzero target with name "ubuntu". `proxy_target` is null if not found
+  proxy_target = one([
+    for each in data.bastionzero_bzero_targets.example.targets
+    : each if each.name == "ubuntu"
+  ])
+}
+
+resource "bastionzero_db_target" "example" {
+  name            = "example-psql-db-target"
+  remote_host     = "localhost"
+  environment_id  = local.env.id
+  remote_port     = 5432
+  proxy_target_id = local.proxy_target.id
+}
+```
+
+### Db target via proxy environment
+
+Create a Db target with the default authentication configuration
+(non-passwordless) and a proxy environment. When a user connects to this target,
+the Bzero or Cluster target with the least number of open connections in this
```

Comment:
Gets a little wonky here since we still call it a proxy target, but in theory I think it should be proxy agent... Maybe we can just say "Linux, Windows, or Kubernetes target"